src/ViewItem.js

```diff
@@ -179,6 +189,27 @@ class ViewItem extends React.Component {
     }
   }
 
+  async componentDidUpdate(prevProps) {
+    if (prevProps.resources.items && this.props.resources.items.records[0] !== prevProps.resources.items.records[0]) {
+      const { lastCheckIn } = this.props.resources.items.records[0];
+
+      if (lastCheckIn) {
+        const { staffMemberId, servicePointId } = lastCheckIn;
+
+        const [staffMembers, servicePoints] = await Promise.all([
```

Comment:
Please try to take advantage of params prop in the manifest in order to avoid the call in the `componentDidUpdate`. See the [example](https://github.com/folio-org/ui-agreements/blob/master/src/routes/AgreementViewRoute.js#L56) for reference.
 @doytch, could you please suggest whether it makes sense?

Reply:
Yeah I think that makes sense @VictorSoroka1. I've definitely been much happier with the functional `params` approach over this `componentDidUpdate` approach in our apps. Would let you avoid a lot of the logic here in this case and make things easier to debug and change in the future.